flybywiresim/a32nx
# fix: Update flaps in flight model for MSFS update 1.14.5.0

Fixes #3913 

## Summary of Changes

**REVISED***
1. Changed flap entries to be compatible with MSFS update 1.14.5.0
2. Improved pitch angles for intermediate flap settings 
3. Reduced maximum N1 to provide more realistic takeoff N1 values
4. Reduced thrust at takeoff N1 and Mach to address comment from @Watsi01 below
5. Increased thrust at approach N1 and speed to provide more realistic N1 values at approach
6. Tailored flap drag to better match N1 values in unreliable airspeed table (which wasn't even attempted with the previous flight/engine mode), provide more accurate approach speed/path control, and reduce landing float that received a lot of comments below.
## Screenshots (if necessary)



## References

Here are the target pitch angles for flaps down conditions:
![Screenshot (697)](https://user-images.githubusercontent.com/70166617/110557120-c9be4e80-80f4-11eb-8e92-41cb0406c2ce.png)

Here were the pitch angles that were achieved in the previous flight model:
![Screenshot (699)](https://user-images.githubusercontent.com/70166617/110557163-e35f9600-80f4-11eb-881a-e2942a16053c.png)
Here are the pitch angles and N1 values achieved with this PR:
![Screenshot (718)](https://user-images.githubusercontent.com/70166617/111024093-b1b72b00-8391-11eb-8d02-a35dfe7c0196.png)
(See discussion below for the reason the CONF 1 pitch angles are not matched. For CONF 3, I placed more emphasis on matching the pitch angles during approach then during level off.)





## Additional context

An issue that we did not have before MSFS SU3 came to light during the development of this PR that limits the accuracy we can achieve with CONF 1 and 1+F lift and drag. As of SU3, we can no longer differentiate lift and drag in the flight model between these 2 configurations. So, not only can we not accurately model slats aerodynamically, we are now limited to having the same lift and drag for CONF 1 as CONF 1+F. I've tried my best to strike a balance between the two.

This ended up being a major rewrite of the flaps portion of the flight model instead of just an MSFS SU3 compatibility fix. It should be much more accurate than the previous flight model in that regard. But it needs more QA testing because of that to make sure there aren't any significant hidden issues. 

One known issue is the landing float issue brought up by many commenters on the first iteration of this PR. I think it has been reduced, but not eliminated. I tested a reduction in the ground effect lift that seemed to fix it pretty well, but it did not work well with the custom fbw version of the mod. So I left it out. Based on that, I think the remaining float issue with the dev division is more of an issue with having an incorrect and inappropriate flight control law active (and even that is not modeled all that well) in the default fbw system used by the dev and stable versions.


Discord username (if different from GitHub):donbikes#4084

## Testing instructions

This is a flight model change affecting flap lift and drag and takeoff and approach thrust/N1. Takeoffs and landings (including approaches) should be conducted in all approved takeoff (1+F, 2, and 3) and landing (3 and Full) configurations. It does not affect flaps up flight characteristics, so full flights are not necessary. 
Specifically look for:
1) The initial takeoff climb performance is modeled to hold approx V2+10 at a pitch angle of 15-18 degrees. Depending on your weight and pitch rate, you may overshoot V2+10, but it should hold speed at the 15-18 degree pitch angle.  It should be vastly improved over iteration 1, but I could only do a limited number of takeoffs at a limited number of locations.
2) Check pitch angles and N1 values in accordance with the tables above, but I am pretty confident that those are okay. Remember that CONF 1 and 1+F have the same lift and drag, so it is not possible to match pitch angles and N1 values precisely for those configurations.
3) Check approach path and speed control in the different flap and landing gear configurations. Because of the CONF 1/CONF 1+F MSFS issue, there is probably a bit more drag in CONF 1+F than IRL. Deceleration capability in CONF 2 is my interpretation of what I am hearing from our pilots -- i don't have any hard data to try to match.
4) Check to see whether the landing characteristics are adequate using the recommended procedures for the airplane or whether additional tweaking is needed. (Keep in mind the comments above regarding differences between the custom and default fbw systems).

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
